cmd/secrets-provider/main.go

```diff
@@ -35,7 +35,7 @@ func main() {
 	}
 
 	// verify we don't run with a K8s secrets storage configuration in a side car
-	if secretsConfig.StoreType == secretsConfigProvider.K8S && authnConfig.ContainerMode != "init" {
+	if secretsConfig.StoreType == secretsConfigProvider.K8S && (authnConfig.ContainerMode != "init" && authnConfig.ContainerMode != "application") {
```

Comment:
do you want to change it to `secretsConfig.StoreType == secretsConfigProvider.K8S && authnConfig.ContainerMode == "sidecar"`?

it is what is written in the comment above (and we can remove the comment)

Reply:
i will remove the comment because I want to be very clear with what types of input we accept

Reply:
and what will happen if we add another container mode in the future? will we add it to this `if` clause too? if we are trying to not support k8s store-type alongside a side-car container mode let's do that.

Reply:
If it is anything else but sidecar, we won't catch it. I want to fail here when it is anything except init or application.

Reply:
not quite. what we are trying to do, as mentioned in the comment, is to `verify we don't run with a K8s secrets storage configuration in a side car`.

prior to your change, container_mode that is not `init` was `sidecar` and that is why we wrote the `if` condition as it is. However, i think that it was written in a wrong way and that it should have been `if secretsConfig.StoreType == secretsConfigProvider.K8S && authnConfig.ContainerMode == "sidecar"`